From 82e2a807e0ebafeff85dee42ea8604986613df89 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 13:54:03 +0700
Subject: [PATCH 257/504] ARM: dts: r8a77470: Add Video Processing Unit Cache
 (VPC module)

This patch support Renesas VPC kernel module probe.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 48 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index a4c025e..cba305f 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -799,6 +799,54 @@
 			power-domains = <&cpg_clocks>;
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				<0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A77470_CLK_VCP0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A77470_CLK_VPC0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A77470_CLK_VPC1>; */
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A77470_CLK_VPC0>; */
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A77470_CLK_VPC1>; */
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		cmt0: timer@ffca0000 {
 			compatible = "renesas,cmt-48-r8a77470",
 				     "renesas,cmt-48-gen2";
-- 
2.7.4

